### ISSUE TITLE
`Pressable` discards `onPressIn` prop when renderer

### ISSUE THREAD

**Issue by jonathanj** [callstack/react-native-testing-library#1676]
## Describe the bug

Rendering `<Pressable onPressIn={…} />` discards the `onPressIn` prop (and `onPressOut` and `onHoverIn` and `onHoverOut`), making it impossible to fire an event for it. If I add a nonsense prop such as `onSomething` that appears as a prop to `View` in the debug output and I can fire an event for it.

If I render a `<View>` instead of `<Pressable>` then there is no issue, but I don't understand why.

I'm not sure if there's something I'm doing wrong, but I did try removing all my Jest setup/RN mocks too, and nothing seemed to help.

```
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function onBlur]}
      onClick={[Function onClick]}
      onFocus={[Function onFocus]}
      onResponderGrant={[Function onResponderGrant]}
      onResponderMove={[Function onResponderMove]}
      onResponderRelease={[Function onResponderRelease]}
      onResponderTerminate={[Function onResponderTerminate]}
      onResponderTerminationRequest={[Function onResponderTerminationRequest]}
      onStartShouldSetResponder={[Function onStartShouldSetResponder]}
      testID="component"
    />
```

## Expected behavior

`onPressIn` (and `onPressOut` and `onHoverIn` and `onHoverOut`) props should be visible on the resulting view.

## Steps to Reproduce

```tsx
import {render, screen} from '@testing-library/react-native';
import {Pressable} from 'react-native';

test('has pressIn', () => {
  render(<Pressable onPressIn={() => null} testID='component' />);
  screen.debug();
  expect(screen.getByTestId('component').props).toHaveProperty('onPressIn');
});
```

## Versions

```
react@npm:18.2.0
react-native@npm:0.74.5
react-test-renderer@npm:18.2.0
@testing-library/react-native@npm:12.7.2
```

**Comment by jonathanj** [comment]
Thanks for the quick response @mdjastrzebski!

> TLDR:
> 
> 1. Your code has an error, you should use `jest.fn()` instead of `jest.mock()`. With this change the test passes for me.

Sorry about that, this was a last minute addition to the issue.

> 2. I would recommend you to use  UserEvent `press()` which offers much more realistic event simulation.

True, although in this case I specifically want to test the behaviour of `onPressIn` and `onHoverIn`, rather than the full user interaction.

> A handful more technical details to help you understand what is happening:
> 
> 1. `Pressable` is a composite component rendering a host `View` element with various props. That `View` does not have an `onPress`, an `onPressIn`, etc event handlers, instead it has a `onResponderGrant` and a `onResponderRelease` handlers.

Thanks, this was helpful and helped me track down the issue. You are correct that my sample does pass, unfortunately distilling down my more complex use case to the sample missed an important detail: The pressable was disabled!

Thanks again for the reply and the help.
